core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1006,6 +1005,10 @@ private void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, Ex
 
             // Update log start offset with the computed value after retention cleanup is done
             remoteLogRetentionHandler.logStartOffset.ifPresent(offset -> handleLogStartOffsetUpdate(topicIdPartition.topicPartition(), offset));
+
+            for (RemoteLogSegmentMetadata segmentMetadata : segmentsToDelete) {
+                remoteLogRetentionHandler.deleteRemoteLogSegment(segmentMetadata, x -> true);
```

Comment:
Can we `and` the result of `remoteLogRetentionHandler.deleteRemoteLogSegment` for all the segments and log an `error` statement if it is not able to delete the segments?